src/task-actions.ts

```diff
@@ -164,7 +165,12 @@ export async function starknetCompileCairo1Action(
     args: TaskArguments,
     hre: HardhatRuntimeEnvironment
 ) {
-    const binDirPath = getCompilerBinDir(args, hre.config.starknet);
+    let binDirPath = getCompilerBinDir(args, hre.config.starknet);
+    if (!binDirPath) {
+        const downloader = new CairoCompilerDownloader(hre.config.paths.root, hre.config.starknet);
```

Comment:
I don't think we should be storing the compilers in the project root. How does hardhat handle this with solidity?

Benefits of not storing in the project:
- Can be used across projets if the users has multiple on their machine (no need to download per project)
- The user won't have to edit their .gitignore to include the binaries